ballerina/memory.bal

```diff
@@ -38,17 +38,97 @@ type MemoryChatSystemMessage readonly & record {|
     *ChatSystemMessage;
 |};
 
+const string SUMMARY_PREFIX = "Summary of previous interactions:";
+final readonly & string:RegExp CHAT_HISTORY_REGEX = re `\{\{CHAT_HISTORY\}\}`;
+final readonly & string:RegExp MAX_SUMMARY_TOKEN_COUNT_REGEX = re `\{\{MAX_SUMMARY_TOKEN_COUNT\}\}`;
+
+final readonly & Prompt DEFAULT_SUMMARY_PROMPT = `
+    You are an expert at summarizing conversations.
+    You will summarize a chat history between users and an AI agent to create a concise summary 
+    that preserves the most important information while adhering to a specified token limit.
+
+    Here is the chat history to summarize:
+
+    <chat_history>
+    {{CHAT_HISTORY}}
+    </chat_history>
+
+    Your summary must contain no more than {{MAX_SUMMARY_TOKEN_COUNT}} tokens.
+
+    Before writing your summary, use the scratchpad below to plan your approach:
+
+    <scratchpad>
+    In this section, think through:
+    - What are the main topics, questions, or issues discussed in the chat history?
+    - What key information, decisions, or conclusions should be preserved?
+    - What details can be omitted or condensed?
+    - How can you structure the summary to be most useful for future reference?
+    </scratchpad>
+
+    When creating your summary, follow these guidelines:
+
+    **What to prioritize:**
+    - The most recent user request or question to ensure the summary reflects the immediate context.
+    - Key questions asked and answers provided
+    - Important decisions made or conclusions reached
+    - Critical context that affects ongoing conversations
+    - Unresolved issues that may need follow-up
+    - Specific facts, numbers, or details that are likely to be referenced again
+
+    **What to minimize or omit:**
+    - Repetitive exchanges or clarifications
+    - Casual conversation or pleasantries
+    - Detailed explanations that can be inferred from context
+    - Tangential discussions that don't affect the main topics
+
+    **Structure your summary:**
+    - Use clear, concise sentences
+    - Group related topics together
+    - Use bullet points or numbered lists when appropriate for clarity
+    - Maintain chronological order when the sequence of events matters.
+
+    Please do not include the scratchpad in your final summary.
+    Please do not include any text before or after the summary.
+`;
+
+# Configuration for summarizing short-term memory content when it overflows.
+public type SummarizeOverflowConfig record {|
```

Comment:
Added a comment on the doc about summarize vs summarization inconsistency.

Anyway, shouldn't this be `OverflowSummarizationConfig`?